tests/components/ActionButton/StatefulActionButton.spec.ts

```diff
@@ -0,0 +1,241 @@
+        let initialState: StatefulActionButtonState;
+        let initialOnEntrySpy: SinonSpy;
+
+        beforeEach(() => {
+            [initialState, initialOnEntrySpy] = createSpiedState('initialState');
+        });
+
+        [
+            {
+                describe: 'if states is not defined',
+                beforeEach: () => {
+                    testSubject = new StatefulActionButton(document.createElement('div'), { states: undefined, initialState: initialState });
+                },
+                expectedWarningMessage: 'The stateful action button cannot render if no states are defined.',
+            },
+            {
+                describe: 'if states is empty',
+                beforeEach: () => {
+                    testSubject = new StatefulActionButton(document.createElement('div'), { states: [] as any, initialState: initialState });
+                },
+                expectedWarningMessage: 'The stateful action button cannot render if no states are defined.',
+            },
+            {
+                describe: 'if initialState is not defined',
+                beforeEach: () => {
+                    testSubject = new StatefulActionButton(document.createElement('div'), { states: [initialState], initialState: undefined });
+                },
+                expectedWarningMessage: 'The stateful action button cannot render if the initial states is not defined.',
+            },
+            {
+                describe: 'if states does not include initialState',
+                beforeEach: () => {
+                    testSubject = new StatefulActionButton(document.createElement('div'), {
+                        states: [createSpiedState('notInitialState')[0]],
+                        initialState: initialState,
+                    });
+                },
+                expectedWarningMessage: 'The stateful action button cannot render if the initial state is not in the list of states.',
+            },
+            {
+                describe: 'if first allowedTransitions contains a transition with a origin state not included in options.state',
+                beforeEach: () => {
+                    testSubject = new StatefulActionButton(document.createElement('div'), {
+                        states: [initialState],
+                        initialState: initialState,
+                        allowedTransitions: [{ from: createSpiedState('someState')[0], to: initialState }],
+                    });
+                },
+                expectedWarningMessage:
+                    'The stateful action button cannot render if one of its transition used a state that is not in the list of states:\n\tOrigin of Transition #0 is not in the list of states. Ensure to use the same object reference as in the options.states.',
+            },
+            {
+                describe: 'if second allowedTransitions contains a transition with a origin state not included in options.state',
```

Comment:
I am not sure about the value of performing the same test twice. Is it simply to ensure the tested code loops through all transitions?

If so, then I would suggest to keep only the test

    if second allowedTransitions contains a transition with a origin state not included in options.state

and rename it

    if allowedTransitions contains a transition with an origin state not included in options.state

Reply:
Yeah I don't remember why I pained myself to do this actually.
😕 will remove